getParameterProfiles.m

```diff
@@ -69,178 +69,198 @@
-                        currentOptions.profile_integ_index = j;
-                        [parameters, fh] = getParProfilesByIntegration(parameters, objective_function, currentOptions, fh);
-                    elseif sum(j == options.profile_integ_index) == 0
-                        currentOptions.profile_optim_index = j;
-                        [parameters, fh] = getParProfilesByOptimization(parameters, objective_function, currentOptions, fh);
-                    else
-                        error('Some really strange error for the profile calculation indices occured');
-                    end
+            otherwise
+                error('Unknown profile computationg method');
+        end
+        options.parameter_index = sort(unique([options.profile_optim_index options.profile_integ_index]));
+    else
+        switch options.profile_method
+            case 'optimization'
+                options.profile_optim_index = options.parameter_index;
+                if ~isempty(options.profile_integ_index)
+                    error('Some profiles seem to be computed twice. Please redefine consistent options!');
                 end
-                
-            elseif strcmp(options.comp_type,'parallel')
-                parfor j = options.parameter_index
-                    currentOptions = options.copy();
-                    if sum(j == options.profile_integ_index) == 1
-                        currentOptions.profile_integ_index = j;
-                        getParProfilesByIntegration(parameters, objective_function, currentOptions, fh);
-                    elseif sum(j == options.profile_integ_index) == 0
-                        currentOptions.profile_optim_index = j;
-                        getParProfilesByOptimization(parameters, objective_function, currentOptions, fh);
-                    else
-                        error('Some really strange error for the profile calculation indices occured');
-                    end
+
+            case 'integration'
+                options.profile_integ_index = options.parameter_index;
+                if ~isempty(options.profile_optim_index)
+                    error('Some profiles seem to be computed twice. Please redefine consistent options!');
                 end
-                
-                % Output
-                switch options.mode
-                    case 'visual', fh = plotParameterProfiles(parameters,'1D',fh,options.parameter_index,options.plot_options);
-                    case 'text' % no output
-                    case 'silent' % no output
+
+            case 'mixed'
+                if (length(unique([options.profile_optim_index, options.profile_integ_index])) ~= length([options.profile_optim_index, options.profile_integ_index])) ...
+                        || (length([options.profile_optim_index, options.profile_integ_index]) ~= length(options.parameter_index))
+                    error('Inconsistent settings for indices in profile calculation.');

```

Comment:
"Inconsistent" won't help you fixing the error, say that numer of (unique) arguments does not agree.

Reply:
You're right, the current code is messy...
I changed to whole thing to something more logic...